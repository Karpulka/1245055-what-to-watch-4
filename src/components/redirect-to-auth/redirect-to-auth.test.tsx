import * as React from "react";
import * as renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import NameSpace from "../../reducer/name-space";
import {StaticRouter} from "react-router-dom";
import RedirectToAuth from "./redirect-to-auth";

const mockStore = configureStore([]);

it(`Render Redirect To Auth AUTH`, () => {
  const store = mockStore({
    [NameSpace.USER]: {
      authorizationStatus: `AUTH`
    }
  });

  const tree = renderer
    .create(<Provider store={store}>
      <StaticRouter>
        <RedirectToAuth>
          <div />
        </RedirectToAuth>
      </StaticRouter>
    </Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Render Redirect To Auth NO_AUTH`, () => {
  const store = mockStore({
    [NameSpace.USER]: {
      authorizationStatus: `NO_AUTH`
    }
  });

  const tree = renderer
    .create(<Provider store={store}>
      <StaticRouter>
        <RedirectToAuth>
          <div />
        </RedirectToAuth>
      </StaticRouter>
    </Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
