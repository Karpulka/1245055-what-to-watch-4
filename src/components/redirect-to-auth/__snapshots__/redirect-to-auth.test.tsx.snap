// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Redirect To Auth AUTH 1`] = `<div />`;

exports[`Render Redirect To Auth NO_AUTH 1`] = `null`;
